### PR TITLE
Add webview to run CRC

### LINKED ISSUE #1235: Webview page to run Code Ready Containers
This page which create a webview interface providing the feature for the user to Run Code Ready Containers.

When the user clicks on `Add OpenShift Cluster` icon on the title panel, the webview is loaded. 
Once the view is active, it shows 2 options to Add an OpenShift Cluster.

1. Local Installation
- Install on Laptop: Red Hat CodeReady Containers.
- Action: Load the CRC view to configure parameters and start crc.

2. Hybrid Installation
- Install, register, and manage OpenShift 4 clusters.
- Action: Redirects to the official document page at https://cloud.redhat.com/openshift/install

These will be shown as cards and respective actions will be enabled on the card button

After box 1 is selected CRC Install page is activated. The page will provide steps to:
1. Select CRC version grouped by OpenShift version
2. Download CRC binary and provide the full path of the crc executable
3. Specify the pull secret
4. Set optional configuration like memory and cores
5. Start Cluster
6. Open Console Url

# Features needed for CRC integration in VSCode #
- [x] Add progress bar for crc command completion

- [x] Run `crc setup` and then `crc start` command as child process

- [x] Show console url once the cluster is up and running

- [x] Showcase command output stream in UI

- [x] Download link to the specific version of crc selected (see beginning of discussion in #1546).

### PR DESCRIPTION
**Pull request by mohitsuman**
Fix #1235

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,6 +61,8 @@
   },
   "dependencies": {
     "@kubernetes/client-node": "^0.11.1",
+    "@material-ui/core": "^4.9.10",
```
Comment (dgolovin) [inline package.json:64]: Please, move react back to dev dependencies.

Comment (mohitsuman) [reply package.json]: Fixed in https://github.com/redhat-developer/vscode-openshift-tools/pull/1520/commits/028b996abb117c745a9e03d01d34f1e277542832

**Comment by dgolovin** [comment]
image
Next is always enabled and usable even without version selected.

**Comment by mohitsuman** [comment]
> image
> Next is always enabled and usable even without version selected.

This is because for crc version, even if the user doesn't select any version. It will always default to `latest` so that download option will have the latest crc to be downloaded. Hence the Next Button is not disabled.

**Comment by mohitsuman** [comment]
> fix UI elements alignment
> image
> Background and boxes location.

Interesting. I have to look for the big screen alignments too here.
I am thinking to remove the `blue background`, what do you suggest ? @dgolovin

File: src/view/cluster/app/clusterView.tsx [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+            case 'crcstatus' :
+              if(message.data === 0) setProgress(true);
+              break;
+            default:
+              break;
+          }
+      }
+    }
+    window.addEventListener('message', messageListener);
+    const crcDownload = `http://mirror.openshift.com/pub/openshift-v4/clients/crc/${versionLabel}/crc-macos-amd64.tar.xz`;
```
Comment (dgolovin) [inline src/view/cluster/app/clusterView.tsx:136]: Always download binaries for macOS.

**Comment by dgolovin** [comment]
> I am thinking to remove the blue background, what do you suggest ? @dgolovin
I leave design choices to you :)

**Comment by jeffmaury** [comment]
I did a first testing following the workflow but did not ask to start the cluster yet. I think the label `Upload pull secret` is confusing, should be rather `Select pull secret file`

File: src/view/cluster/app/clusterView.tsx [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+  { crcVersion: "1.1.0", openshiftVersion: "4.2.2"},
+  { crcVersion: "1.2.0", openshiftVersion: "4.2.8"},
+  { crcVersion: "1.3.0", openshiftVersion: "4.2.10"},
+  { crcVersion: "1.4.0", openshiftVersion: "4.2.13"},
+  { crcVersion: "1.5.0", openshiftVersion: "4.2.14"},
+  { crcVersion: "1.6.0", openshiftVersion: "4.3.0"},
+  { crcVersion: "1.7.0", openshiftVersion: "4.3.1"},
+  { crcVersion: "1.8.0", openshiftVersion: "4.3.8"},
+  { crcVersion: "1.9.0", openshiftVersion: "4.3.10"},
+  { crcVersion: "latest", openshiftVersion: "4.4.3"}
```
Comment (dgolovin) [inline src/view/cluster/app/clusterView.tsx:102]: How do I know what is 'latest' 1.9.1 or 1.10.0? There should be version in label.

Comment (mohitsuman) [reply src/view/cluster/app/clusterView.tsx:102]: I have used the nomenclature what crc has on mirror.openshift.com. So for this latest is 1.10.0. I can make it this way: 
` { crcVersion: "1.10.0(latest)", openshiftVersion: "4.4.3"}`

**Comment by mohitsuman** [comment]
> I did a first testing following the workflow but did not ask to start the cluster yet. I think the label `Upload pull secret` is confusing, should be rather `Select pull secret file`

Fixed in 57916583d7d9bfd69d4fa951752b6cbe72b3a73b

**Comment by mohitsuman** [comment]
> fix UI elements alignment
> image
> Background and boxes location.

@dgolovin This should be fixed for big screens with the commit 57916583d7d9bfd69d4fa951752b6cbe72b3a73b Can you try and check how it looks on big screens ?

File: src/view/cluster/app/clusterView.tsx (hunk 2 above)
Comment (mohitsuman) [reply src/view/cluster/app/clusterView.tsx]: Fixed in commit 83529434748eb4e4fb79ea5fc3e4b5ec13534631

File: src/view/cluster/app/clusterView.tsx (hunk 3 above)
Comment (mohitsuman) [reply src/view/cluster/app/clusterView.tsx]: I have added the latest label to the UI dropdown for crc version select.

**Comment by jeffmaury** [comment]
I did a test on Win10 and it was not successful for me: the last message I can see it:
`level=info msg="Creating CodeReady Containers VM for OpenShift 4.3.10..."`
and using the OS process explorer I could see 2 crc processes launched but after a while they disappear but the message stays the same. Clicking on `Stop Process` has no visible effect

**Pull request merged**